Objective-C/Tests/ReplicatorTest+PendingDocIds.m

```diff
@@ -353,6 +397,44 @@ - (void) testIsDocumentPendingWithPushFilter {
     [replicator removeChangeListenerWithToken: token];
 }
 
+- (void) testIsDocumentPendingWhenOffline {
+    XCTestExpectation* x = [self expectationWithDescription: @"Replicator Stopped"];
+    CBLMutableDocument* doc = [self createDocument: @"doc-1"];
+    [doc setString: kCreateActionValue forKey: kActionKey];
+    [self saveDocument: doc];
+    
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    CBLReplicatorConfiguration* config = [self configWithTarget: target
+                                                           type: kCBLReplicatorTypePushAndPull
+                                                     continuous: YES];
+    CBLReplicator* replicator = [[CBLReplicator alloc] initWithConfig: config];
+    __block CBLReplicator* r;
+    id<CBLListenerToken> token = [replicator addChangeListener: ^(CBLReplicatorChange* change) {
+        r = replicator;
+        NSError* err = nil;
+        if (change.status.activity == kCBLReplicatorConnecting) {
+            Assert([change.replicator isDocumentPending: @"doc-1" error: &err]);
+        } else if (change.status.activity == kCBLReplicatorIdle) {
+            AssertFalse([change.replicator isDocumentPending: @"doc-1" error: &err]);
+            [r setSuspended: YES];
+            
```

Comment:
Yeah... Is this synchronous?  Is there any chance you could get the OFFLINE, before you insert the document

Reply:
It's asynchronous and I think it could be a race condition but pretty difficult to happen. Maybe save the doc before suspend?

Reply:
I was thinking that the test would be clearer if it asserted that the first call definitely was connecting, the second idle, and the third offline.  I agree that it isn't a big deal, though...

Reply:
Save before suspend might also cause, replicator to push the docs before suspending. I will update to avoid the situation.

Reply:
You can save and suspend in batch so that the doc will not be pushed before suspending.

Reply:
restructured,  differently